### PR TITLE
Examples unit test configurability

### PR DESCRIPTION
**Pull request by tenableAndrew**
- Allow example scan tests to be run against a configurable host, not just 'tenable.com'.
- Also changed scan type to basic

### CODE REVIEW

File: tests/integration/test_examples.py [hunk 1]
```diff
@@ -17,7 +17,7 @@ def test_policies(self, app):
 
     def test_scans(self, app):
         from examples.scans import example
-        example(app.session_name, app.session_file_output)
+        example(app.session_name, app.session_file_output, app.session_scan_target)
```
Comment (alanfung) [inline tests/integration/test_examples.py]: It's fine to just pass `TenableIOTestConfig.get('scan_text_targets')` directly here. The other two methods are needed to generate test-session unique names.

File: tests/conftest.py [hunk 2]
```diff
@@ -30,6 +30,9 @@ def session_file_output(self, name, content=None):
                 fd.write('test_file')
         return path
 
+    def session_scan_target(self):
+        return TenableIOTestConfig.get('scan_text_targets')
+
```
Comment (alanfung) [inline tests/conftest.py]: Don't need changes in this file, see comment on `test_examples.py`.

File: examples/scans.py [hunk 3]
```diff
@@ -26,8 +27,8 @@ def example(test_name, test_file):
     '''
     scan = client.scan_helper.create(
         name=scan_name,
-        text_targets='tenable.com',
-        template='discovery'
+        text_targets=scan_target,
+        template='basic'
```
Comment (alanfung) [inline examples/scans.py]: I am not sure if it's necessary to run `'basic'` here. If `'discovery'` is faster than `'basic'` in anyway, we should keep it as `'discovery'` because it is sufficient for the purpose of examples and not having to add additional time to the already slow test-runs.

File: examples/scans.py [hunk 4]
```diff
@@ -9,12 +9,13 @@
 from tenable_io.exceptions import TenableIOApiException
 
 
-def example(test_name, test_file):
+def example(test_name, test_file, test_target):
```
Comment (alanfung) [inline examples/scans.py]: Should name this `test_targets` for consistency.

File: tests/conftest.py [hunk 5]
```diff
@@ -7,7 +7,7 @@
 from tests.util import upload_image
 
 from tenable_io.client import TenableIOClient
-
+from tests.config import TenableIOTestConfig
 
 class App:
```
Comment (alanfung) [inline tests/conftest.py]: `tests/conftest.py:12:1: E302 expected 2 blank lines, found 1` per pep8.

File: examples/scans.py (hunk 3 above)
Comment (tenableAndrew) [reply examples/scans.py]: My only concern was that 'discovery' scans typically only return INFO level vulnerabilities which are not displayed in the Workbench by default. I wasn't sure whether any of the other tests were dependent on data being available there.

File: tests/conftest.py (hunk 5 above)
Comment (tenableAndrew) [reply tests/conftest.py]: Are you using a pylintrc file? This did not appear when I ran pylint in python3. I've updated all the files I touched to be two lines before the start of a class.

File: examples/scans.py (hunk 3 above)
Comment (alanfung) [reply examples/scans.py]: There is no dependency to this examples script, it is solely for demonstration only. It isn't to test the code, but to test that the examples are not outdated.

Comment (tenableAndrew) [reply examples/scans.py]: I'll change back to `'discovery'`, it's much faster than basic.

Comment (tenableAndrew) [reply examples/scans.py]: I'm having trouble to get this to pass in our engineering environment with the template being set to `'discovery'`, for a single target without auth, a basic scan really shouldn't take too much longer.

File: tests/conftest.py (hunk 5 above)
Comment (alanfung) [reply tests/conftest.py]: We don't have a `pylintrc` file setup. We run `pep8 tenable_io tests` (you can see that in `.travis.yml`).

File: tests/conftest.py [hunk 6]
```diff
@@ -1,7 +1,7 @@
 import os
-import pytest
 import shutil
 import uuid
+import pytest
```
Comment (alanfung) [inline tests/conftest.py]: Looks like unnecessary change on this file.
